packages-exp/auth-exp/test/integration/webdriver/util/functions.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/** Available in the browser. See static/anonymous.js */
+export enum AnonFunction {
+  SIGN_IN_ANONYMOUSLY = 'anonymous.anonymous'
+}
+
+/** Available redirect functions. See static/redirect.js */
+export enum RedirectFunction {
+  IDP_REDIRECT = 'redirect.idpRedirect',
+  IDP_REAUTH_REDIRECT = 'redirect.idpReauthRedirect',
+  IDP_LINK_REDIRECT = 'redirect.idpLinkRedirect',
+  REDIRECT_RESULT = 'redirect.redirectResult',
+  GENERATE_CREDENTIAL_FROM_RESULT = 'redirect.generateCredentialFromRedirectResultAndStore',
+  SIGN_IN_WITH_REDIRECT_CREDENTIAL = 'redirect.signInWithRedirectCredential',
+  LINK_WITH_ERROR_CREDENTIAL = 'redirect.linkWithErrorCredential',
+  CREATE_FAKE_GOOGLE_USER = 'redirect.createFakeGoogleUser',
+  TRY_TO_SIGN_IN_UNVERIFIED = 'redirect.tryToSignInUnverified'
+}
+
+/** Available core functions within the browser. See static/core.js */
+export enum CoreFunction {
+  RESET = 'core.reset',
+  AWAIT_AUTH_INIT = 'core.authInit',
+  USER_SNAPSHOT = 'core.userSnap',
+  AUTH_SNAPSHOT = 'core.authSnap',
+  SIGN_OUT = 'core.signOut'
```

Comment:
Thoughts on exposing `auth` functions directly? Maybe as `AuthFunction.signOut`? You can even use TypeScript type mapping to make it easier

Reply:
There are only a few functions that would work that way (taking no parameters). The serialization messes up the cross-communication